### PR TITLE
cleanup fireRequests and remove axios

### PR DESCRIPTION
**Pull request by sainak**
### <samp>🤖 Generated by Copilot at ca617a1</samp>

This pull request replaces `axios` with `fetch` for making HTTP requests throughout the codebase, and removes the `axios` dependency from the `package.json` file. This improves the performance, compatibility, and consistency of the HTTP requests and reduces the code complexity. It also fixes some minor issues with the routes and authentication in the `api.tsx` file.

## Proposed Changes

- Replace axios with native fetch

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖 Generated by Copilot at ca617a1</samp>

*  Remove axios dependency and import statements from various files ([link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L65), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2L14), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-9c97aeeffdf95a04b24615b5a0842918a82f50040f9e09ca7ec69ceac2ea759bL2), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-1d65a996f952fb6cd0b8721be6e07b3b0bc719c35adb7ddd256b3e550a02a7dbL11), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-6352cfe4ad14fd5cf49cc8bca3a81880e50c91515c2c11bdff1b379a587c4983L17), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-d139b2498abb36fc13251b14481a9b86a6f7b2efc6a4b8427b417d1f363981b5L1), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-6624210f1e9cdff909251ba965fafd78918e3d0a95fb569b8f151e8528eca8b6L1), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-de00ad6b21764a6d88a0a1acf9080679a9042f2c4a88b704c65ae62e18068023L2))
*  Replace axios with fetch for making HTTP requests in various files, using custom helper functions and adjusting the request parameters, headers, and body as needed ([link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-26ad4b834941d9b19ebf9db8082bd202aaf72ea0ddea85f5a8a0cb3c729cc6f2L43-R56), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-9c97aeeffdf95a04b24615b5a0842918a82f50040f9e09ca7ec69ceac2ea759bL62-R68), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-1d65a996f952fb6cd0b8721be6e07b3b0bc719c35adb7ddd256b3e550a02a7dbL113-R120), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-819767f9fa040d9a0f1489842f6e34b51f5e017e5cc2f23073025fcd68c95412L1-R1), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-819767f9fa040d9a0f1489842f6e34b51f5e017e5cc2f23073025fcd68c95412L9-R9), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-819767f9fa040d9a0f1489842f6e34b51f5e017e5cc2f23073025fcd68c95412L22-R22), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-d139b2498abb36fc13251b14481a9b86a6f7b2efc6a4b8427b417d1f363981b5L113-R115), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-d139b2498abb36fc13251b14481a9b86a6f7b2efc6a4b8427b417d1f363981b5L122-R123), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-6624210f1e9cdff909251ba965fafd78918e3d0a95fb569b8f151e8528eca8b6R34), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-6624210f1e9cdff909251ba965fafd78918e3d0a95fb569b8f151e8528eca8b6L931-R934), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-6624210f1e9cdff909251ba965fafd78918e3d0a95fb569b8f151e8528eca8b6L1052-R1038), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-de00ad6b21764a6d88a0a1acf9080679a9042f2c4a88b704c65ae62e18068023L12-R11), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-de00ad6b21764a6d88a0a1acf9080679a9042f2c4a88b704c65ae62e18068023R20), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-de00ad6b21764a6d88a0a1acf9080679a9042f2c4a88b704c65ae62e18068023L108-R124), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-a0c88ccc2116ceb542bc9be228c54ed98b0bc89e2d9daf66c941f1a0305e626eL6-R11))
*  Remove unused or redundant actions and error checks related to axios ([link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-6352cfe4ad14fd5cf49cc8bca3a81880e50c91515c2c11bdff1b379a587c4983L96-R97), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-a0c88ccc2116ceb542bc9be228c54ed98b0bc89e2d9daf66c941f1a0305e626eL726-L729))
*  Add trailing slashes to various routes in `api.tsx` to match the backend expectations ([link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL26-R32), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL103-R104), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL113-R114), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL152-R153), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL169-R175), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL287-R288), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL391-R402), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL418-R422), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL483-R478), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL551-R546), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL567-R562), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL574-R581), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL602-R597), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL625-R620), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL635-R634), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL708-R703), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL747-R746), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL777-R772), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL816-R811), [link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL898-R893))
*  Add a noAuth flag to the token_refresh route in `api.tsx` to indicate that it does not require authentication ([link](https://github.com/coronasafe/care_fe/pull/6003/files?diff=unified&w=0#diff-21639e8b3b95469edd34a3d2641a85e6eb02393c442301e11d91665f10f1cf7cL26-R32))

### CODE REVIEW

**Comment by nihal467** [comment]
@sainak check the cypress issue

**Comment by khavinshankar** [comment]
@rithviknishad @Ashesh3 can you please review this pr once

**Review by Ashesh3 — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

File: src/Redux/fireRequest.tsx [hunk 1]
```diff
@@ -47,29 +45,30 @@ export const fetchResponseSuccess = (key: string, data: any) => {
...
     }
-    isRunning[altKey ? altKey : key] = axios.CancelToken.source();
+    const controller = new AbortController();
+    isRunning[requestKey] = controller;
+
     // get api url / method
     const request = Object.assign({}, requestMap[key]);
-    if (path.length > 0) {
-      request.path += "/" + path.join("/");
-    }
```
Comment (rithviknishad) [inline src/Redux/fireRequest.tsx]: Add this back. Things like update facility and all is not working as facility id and all is passed to `path` argument in the action.

Comment (rithviknishad) [reply src/Redux/fireRequest.tsx]: As you can see, the facility id is missing from the PUT request.

image

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]
Recent changes approved on behalf of @sainak

**Comment by nihal467** [comment]
@sainak 
image

https://deploy-preview-6003--care-net.netlify.app/facility/45d45bb6-c48e-4883-984d-ac602ef7df1e/patient/5ea5550d-acc3-426e-888f-42f0ed58b705/consultation/debbce92-7670-45c0-93e4-82cd92241a6c/feed

- [ ] when I try to move a camera to a fixed preset, the moving symbol stuck in the loading state post moving to the preset position. In existing staging, there is no issue like that.

Note: This issue was only observed while moving to a preset, but manual controls are working fine

image

- [ ] unable to upload a facility cover image

image

- [ ] critical care page is not at all loading 

https://deploy-preview-6003--care-net.netlify.app/facility/45d45bb6-c48e-4883-984d-ac602ef7df1e/patient/5ea5550d-acc3-426e-888f-42f0ed58b705/consultation/debbce92-7670-45c0-93e4-82cd92241a6c/daily_rounds/4a42ec26-3dd0-4998-8b6f-4fbd9ba96e27/update

CC: @rithviknishad

**Comment by nihal467** [comment]
@sainak any ETD for this PR status, as its been stale with no activity for 3 weeks ?

**Comment by Ashesh3** [comment]
> @sainak any ETD for this PR status, as its been stale with no activity for 3 weeks ?

^ @sainak 

cc: @nihal467

**Pull request closed without merging**